### PR TITLE
feat(Button): creating button

### PR DESCRIPTION
**Pull request by slhay28**
**JIRA**
LUI-1138

**Description**
Create Button, Button stories, and Button styles

### CODE REVIEW

File: src/packages/solid/components/Button/Button.styles.ts [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+/**
+ * allows the property to either be the supplied type,
+ * or an array containing that type and an animationSettings object
+ */
+type Animatable<Type> = Type | [Type, AnimationSettings];
+
+/**
+ * states can contain any of the style object's properties, except other states
+ */
+type StateStyle<Type> = Partial<Omit<Type, States>>;
```
Comment (anthony9187) [inline src/packages/solid/components/Button/Button.styles.ts:36]: can we move these out to the shared `types` file?

File: src/packages/solid/components/Button/Button.tsx [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+
+  const prefixIcon = props.prefix?.icon ? true : false;
+  //const prefixCheckbox = props.prefix?.checked? true : false;
+
+  const suffixIcon = props.suffix?.icon ? true : false;
+  //const suffixCheckbox = props.suffix?.checked? true : false;
+  return (
+    <node
+      {...props}
+      use:withPadding={styles.Container?.padding}
```
Comment (anthony9187) [inline src/packages/solid/components/Button/Button.tsx:52]: we should move this directive above the props spread
https://github.com/lightning-js/solid#use-directives-in-solid

**Review by anthony9187 — COMMENTED** [review]
looks great so far! two things:
1. can we add states to the story controls? specifically focus
2. the button text looks off center vertically - a bit too high up. can we try and pull that down?

File: src/packages/solid/components/Button/Button.tsx (hunk 2 above)
Comment (chiefcll) [reply src/packages/solid/components/Button/Button.tsx:52]: PS - I think the latest solidjs fixes that - but good to be safe

**Pull request merged**